server/src/interfaces/access.interface.ts

```diff
@@ -20,7 +22,7 @@ export interface IAccessRepository {
 
   album: {
     checkOwnerAccess(userId: string, albumIds: Set<string>): Promise<Set<string>>;
-    checkSharedAlbumAccess(userId: string, albumIds: Set<string>): Promise<Set<string>>;
+    checkSharedAlbumAccess(userId: string, albumIds: Set<string>, access: AlbumUserRole): Promise<Set<string>>;
```

Comment:
Nit: What do you think of renaming `access` to `role`? To me this fits better into the general naming you chose

Reply:
I gave this a different name intentionally, because this parameter will be used a little differently than other places. This is to communicate that we are not looking for that exact role, but rather Viewer access (Viewer and Editor roles) or Editor access (only Editor role)